.github/workflows/cleanup.yml

```diff
@@ -0,0 +1,98 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Reusable workflow that handles CLI request cleanup.
+name: 'aod-cleanup'
+
+# Support below trigger:
+# pull_request:
+#   types: 'closed'
+#   paths: 'cli.yaml'
+on:
+  workflow_call:
+    inputs:
+      aod_cli_version:
+        description: 'The version of AOD CLI.'
+        type: 'string'
+        default: 'latest'
+        required: false
+      go_version:
+        description: 'The version of Golang.'
+        type: 'string'
+        default: '1.20'
+        required: false
+
+jobs:
+  approved:
```

Comment:
An much easier way is to check ```if: '${{ github.event.pull_request.mergeable}}'``` instead of this approved job. But it won't work since all AOD requests will have a failure check to prevent them from merging.

Reply:
I will try graphql instead, ref:https://github.com/orgs/community/discussions/24702